modules/http2/h2_mplx.c

```diff
@@ -254,18 +314,20 @@ h2_mplx *h2_mplx_c1_create(h2_stream *stream0, server_rec *s, apr_pool_t *parent
     m->streams_ev_in = apr_array_make(m->pool, 10, sizeof(h2_stream*));
     m->streams_ev_out = apr_array_make(m->pool, 10, sizeof(h2_stream*));
 
-#if !H2_POLL_STREAMS
+    m->streams_input_read = h2_iq_create(m->pool, 10);
     status = apr_thread_mutex_create(&m->poll_lock, APR_THREAD_MUTEX_DEFAULT,
                                      m->pool);
     if (APR_SUCCESS != status) goto failure;
-    m->streams_input_read = h2_iq_create(m->pool, 10);
+#if !H2_POLL_STREAMS
```

Comment:
Can't we do the same thing for output streaming, i.e. wakeup pollset instead of pipe putc in `c2_beam_output_write_notify()`?
And then get rid of `H2_POLL_STREAMS`, but I may be missing something..

Reply:
I thought about this. We have support for disabling this bc Windows and older APR versions.

The possible benefit of the file descriptors is that we might adapt `mod_proxy` modules to use them for shuffling content back and forth, as you did for the wstunnel.

But then again, as long as we do not have that on all platforms we support, this means different code paths and complexity.

Reply:
Oh, I wasn't thinking of removing file descriptors but wondered how you managed to remove the `H2_POLL_STREAMS` support for `streams_input_read` and why you couldn't do the same for the output side.